File: mobile/lib/features/bookstore/domain/entities/novel.dart
class Novel {
  final String id;
  final String title;
  final String authorName;
  final String? coverUrl;
  final String? description;
  final String status;
  final String language;
  final int totalWords;
  final int totalChapters;
  final int viewCount;
  final int favoriteCount;
  final double avgRating;
  final String? latestChapterTitle;
  final DateTime? latestChapterAt;
  final List<String> categories;
  final List<String> tags;

  const Novel({
    required this.id,
    required this.title,
    required this.authorName,
    this.coverUrl,
    this.description,
    this.status = 'ongoing',
    this.language = 'zh-TW',
    this.totalWords = 0,
    this.totalChapters = 0,
    this.viewCount = 0,
    this.favoriteCount = 0,
    this.avgRating = 0,
    this.latestChapterTitle,
    this.latestChapterAt,
    this.categories = const [],
    this.tags = const [],
  });
}

File: web/types/novel.ts
export interface Novel {
  id: string
  title: string
  author_name: string
  cover_url: string
  description: string
  status: 'ongoing' | 'completed' | 'hiatus'
  language: 'zh-TW' | 'zh-CN' | 'en'
  total_words: number
  total_chapters: number
  view_count: number
  favorite_count: number
  avg_rating: number
  latest_chapter_title: string
  latest_chapter_at: string
  categories: Category[]
  tags: Tag[]
}

export interface Category {
  id: number
  name: string
  slug: string
  parent_id?: number
  children?: Category[]
}

export interface Tag {
  id: number
  name: string
}

export interface Chapter {
  id: string
  novel_id: string
  title: string
  content: string
  chapter_number: number
  word_count: number
}
